src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -177,15 +188,22 @@ public void RunAndBlock(CancellationToken cancellationToken = default(Cancellati
                             }, TaskContinuationOptions.ExecuteSynchronously);
                     }
 
-                    // Wait for a short period of time before restarting to
-                    // avoid cases where a host level config error might cause
-                    // a rapid restart cycle
-                    Task.Delay(_config.RestartInterval).GetAwaiter().GetResult();
+                    // attempt restarts using an exponential backoff strategy
+                    CreateRestartBackoffDelay(consecutiveErrorCount).GetAwaiter().GetResult();
```

Comment:
Can we (verbosely) log the number of consecutive errors here? 

I'm thinking about when we try to debug with the logs -- seeing a large delay may be confusing to some. Seeing something that says "Consecutive error count: 3. Backing off restart." may help explain why there's a gap in the timestamps between "A ScriptHost error has occurred." and the actual restart of the host.